### DISCUSSION TITLE
We have updated our project to .net 9 from .net 8 and are hitting a runtime error "WebAssembly.compileStreaming()".

### DISCUSSION THREAD

**Discussion by BradChase2011** [unoplatform/Uno.Wasm.Bootstrap#916] [Q&A]
Has anyone encountered the error after upgrading to .net 9?  At runtime we see this:
![image](https://github.com/user-attachments/assets/0c09d1d9-9111-4b30-ab24-f789c86cc5ba)

MONO_WASM: instantiate_wasm_module() failed CompileError: WebAssembly.compileStreaming(): Compiling function #27382:"CompressionNative_Crc32" failed: expected 1 elements on the stack for fallthru, found 4 @+7891257

I cant seem to find any info on it anywhere.  Anyone know how to get around this or any info at all would really help.  I used this link to get it moved over to .net 9:  https://platform.uno/docs/articles/external/uno.wasm.bootstrap/doc/using-the-bootstrapper.html.

Thanks much!

**Comment by jeromelaban** [comment]
Thanks for the report, this looks like this .NET 9 RC2  issue https://github.com/dotnet/runtime/issues/109289, likely caused by the fact that you're referencing SkiaSharp. There's a temporary workaround listed in the comments.

**Reply by BradChase2011** [reply]
I will pile through it, thanks for the reference.  Yea I just updated our custom SkiaSharp build and noticed they have some issues with references to the native libraries (we have changes in these as well).  Currently we have a hack in our proj to remove and reinject the correct native libs.  That said, we do our own SkiaSharp builds, is there a way to fix it directly in SkiaSharp?  Or does the issue lay elsewhere?  I added the hack you suggested into our proj file but the loading stops now at the "Setting UNO_BOOTSTRAP_WEBAPP_BASE_PATH=./" line and does not hit the main function and there is no error, but  I presume thats not where the target should be placed?  Who or which library will this be fixed in for tracking, is it dotnet runtime?  Thanks much!

**Reply by jeromelaban** [reply]
Thanks for the updates. We do not have those information at this time, you'll find up-to-date details in the issue I mentioned as they get published.

**Reply by BradChase2011** [reply]
Understood.  Ill keep track, thanks much!

**Reply by BradChase2011** [reply]
I have tried to create a sample uno wasm proj using the new "Sdk="Microsoft.NET.Sdk.WebAssembly"" but am unable to ever get the Main function called.  This is just a simple project with no other libraries.  Do you have a sample project that works for you?  Or is the Main function not getting called part of the known bug?  I do get all the setting's "Setting UNO_BOOTSTRAP_" output in the console but not any further.

So I wanted to clarify that the main not being called is the same between the one with skia and the sample without (and without the target).

EDIT:  I know what is causing the main not to load and that's "<WasmShellEnableThreads>true</WasmShellEnableThreads>".  If I set threading to false, then the main will load, true and it breaks.

**Reply by jeromelaban** [reply]
Threading is not working properly at this time, it's a work in progress on Microsoft's side.

**Reply by BradChase2011** [reply]
Ohh shoot ok, I thought through this incorrectly.  I thought the threading was working like with the .net 8.0 side and Microsoft had moved their threading work to .net 10.0.  We mainly need on emscripten 56+.
